### PR TITLE
[NO-TICKET] ActionController - Honor resource override

### PR DESCRIPTION
**Pull request by vpellan**
**What does this PR do?**

Honor resource override in ActionController

**Motivation:**

Internal support

**Change log entry**

Yes. Preserve custom trace resource names for ActionController requests.

**Additional Notes:**

**How to test the change?**

### CODE REVIEW

File: lib/datadog/tracing/contrib/action_pack/action_controller/instrumentation.rb [hunk 1]
```diff
@@ -59,7 +62,9 @@ def finish_processing(payload)
 
               begin
                 # We repeat this in both start and at finish because the resource may have changed during the request
-                trace.resource = span.resource unless payload[:headers][:request_exception]
+                if trace.resource == tracing_context[:dd_request_trace_resource] && !payload[:headers][:request_exception]
```
Comment (mabdinur) [inline lib/datadog/tracing/contrib/action_pack/action_controller/instrumentation.rb:65]: Could we also treat `!trace.resource_override?` as eligible here? `TraceOperation` defines `trace.resource = nil` as clearing the override, but this equality then fails and Rack falls back to its generic resource.
A small spec for that transition plus `!trace.resource_override? || ...` would preserve the existing no-override behavior.

**Review by mabdinur — APPROVED** [review]
The resource-ownership approach matches the existing Rack precedence, and the tests cover overrides both before controller start and during request processing. I left one non-blocking edge-case note.

**Review by TonyCTHsu — CHANGES_REQUESTED** [review]
1. The pull request description does not explain *why*.
2. Changing `resource` has been risky because of backward compatibility, please elaborate how this change impact customers.

**Comment by marcotc** [comment]
I agree with @TonyCTHsu that we need some clarity on what's the behaviour fix intended here.

**Review by marcotc — CHANGES_REQUESTED** [review]
Just asking for description update.

**Comment by vpellan** [comment]
Closing for now, this was handled differently by support
